src/server/scripts/Spells/spell_holiday.cpp

```diff
@@ -277,6 +277,84 @@ class spell_hallow_end_tricky_treat : public SpellScriptLoader
         }
 };
 
+// Hallowed wands
+enum HallowendData
+{
+    //wand spells
+    SPELL_HALLOWED_WAND_PIRATE             = 24717,
+    SPELL_HALLOWED_WAND_NINJA              = 24718,
+    SPELL_HALLOWED_WAND_LEPER_GNOME        = 24719,
+    SPELL_HALLOWED_WAND_RANDOM             = 24720,
+    SPELL_HALLOWED_WAND_SKELETON           = 24724,
+    SPELL_HALLOWED_WAND_WISP               = 24733,
+    SPELL_HALLOWED_WAND_GHOST              = 24737,
+    SPELL_HALLOWED_WAND_BAT                = 24741
+};
+
+class spell_hallow_end_wand : public SpellScriptLoader
+{
+public:
+    spell_hallow_end_wand() : SpellScriptLoader("spell_hallow_end_wand") {}
+
+    class spell_hallow_end_wand_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_hallow_end_wand_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellEntry*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_PIRATE_COSTUME_MALE) ||
+                !sSpellMgr->GetSpellInfo(SPELL_PIRATE_COSTUME_FEMALE) ||
+                !sSpellMgr->GetSpellInfo(SPELL_NINJA_COSTUME_MALE) ||
+                !sSpellMgr->GetSpellInfo(SPELL_NINJA_COSTUME_FEMALE) ||
+                !sSpellMgr->GetSpellInfo(SPELL_LEPER_GNOME_COSTUME_MALE) ||
+                !sSpellMgr->GetSpellInfo(SPELL_LEPER_GNOME_COSTUME_FEMALE) ||
+                !sSpellMgr->GetSpellInfo(SPELL_GHOST_COSTUME_MALE) ||
+                !sSpellMgr->GetSpellInfo(SPELL_GHOST_COSTUME_FEMALE))
+                return false;
+            return true;
+        }
+
+        void HandleScriptEffect()
+        {
+            Unit* caster = GetCaster();
+            Unit* target = GetHitUnit();
+
+            uint32 spellId = 0;
+            uint8 gender = target->getGender();
+
+            switch (GetSpellInfo()->Id)
+            {
+                case SPELL_HALLOWED_WAND_LEPER_GNOME:
+                    spellId = gender ? SPELL_LEPER_GNOME_COSTUME_FEMALE : SPELL_LEPER_GNOME_COSTUME_MALE;
+                    break;
+                case SPELL_HALLOWED_WAND_PIRATE:
+                    spellId = gender ? SPELL_PIRATE_COSTUME_FEMALE : SPELL_PIRATE_COSTUME_MALE;
+                    break;
+                case SPELL_HALLOWED_WAND_GHOST:
+                    spellId = gender ? SPELL_GHOST_COSTUME_FEMALE : SPELL_GHOST_COSTUME_MALE;
+                    break;
+                case SPELL_HALLOWED_WAND_NINJA:
+                    spellId = gender ? SPELL_NINJA_COSTUME_FEMALE : SPELL_NINJA_COSTUME_MALE;
+                    break;
+                case SPELL_HALLOWED_WAND_RANDOM:
+                    spellId = RAND(SPELL_HALLOWED_WAND_PIRATE, SPELL_HALLOWED_WAND_NINJA, SPELL_HALLOWED_WAND_LEPER_GNOME, SPELL_HALLOWED_WAND_SKELETON, SPELL_HALLOWED_WAND_WISP, SPELL_HALLOWED_WAND_GHOST, SPELL_HALLOWED_WAND_BAT);
+                    break;
+            }
+            caster->CastSpell(target, spellId, true);
```

Comment:
well
```
default;
    break;
```
will still cast spellID 0, but if you make it return it won't cast anything
```
default:
    return;
```